### DISCUSSION TITLE
【keyball44】左側OLEDが表示しない

### DISCUSSION THREAD

**Discussion by maikeru0212** [Yowkees/keyball#667] [Q&A]
初めまして、タイトルの現象についてご教授願いたいです。

Keyball44で左側OLEDが映らない状態となっています。
また、左右をTRSで接続しPCとUSB接続した場合
左側USB時：左右両方のOLEDが表示されない
右側USB時：右側OLED表示、左側OLED表示せず
といった状態です。

ファームウェアKeyball61_via_v1.3.2の書き込みは完了
key入力・トラックボール動作はすべて正常

左側のハンダを疑い、目視で確認しましたがショートしていそうな箇所もわからず
困っております。

またテスターもあるので、確認方法があれば併せてご教授頂けると幸いです。

よろしくお願いします。
![IMG_4262 1](https://github.com/user-attachments/assets/56557d52-5ea0-4bd9-8173-092a9dfbed53)
![IMG_4263 1](https://github.com/user-attachments/assets/6a2a2ab5-2608-4071-8d73-d66aed3ca489)
![IMG_4264 1](https://github.com/user-attachments/assets/2cb46d2e-619a-4785-bf03-ea0d59f06ea0)
![IMG_4265 1](https://github.com/user-attachments/assets/dd3710ae-e5b8-49fe-8d91-acf0c53ac3e4)
![IMG_4266 1](https://github.com/user-attachments/assets/f781be28-19a5-4a26-a5ca-5980a6a6ed4b)

**Comment by jtuyen** [comment]
I would suggest cleaning up the flux around the OLED pins as it may be the cause for shorts.

**Reply by maikeru0212** [reply]
ご返信ありがとうございます。
フラックスを除去したんですが、改善されませんでした。
他になにか改善する方法はありませんか？

Thank you for your message
The flux was removed, but the phenomenon was not improved.

Are there any other effective methods?



![IMG_4269 1](https://github.com/user-attachments/assets/52130b05-1b7e-4355-a5ac-3f938f9a9f8f)

**Reply by jtuyen** [reply]
On your left side MCU, check the SDA and SCL pins for short. Mayble also check the VCC and GND for possible short as well.

**Reply by maikeru0212** [reply]
それぞれのピンを確認しましたがショートしてませんでした。
はんだ付けの時に何かしら基盤を破損してしまった可能性もありますかね・・・

I checked each pin and there were no shorts.
Is it possible that the board was somehow damaged during soldering?

**Reply by jtuyen** [reply]
Use a multimeter to run a continuity check between all the OLED pins to MCU pins for faults. If the continuity results are good, use a multimeter to check if the VCC is running 3.3V. If the voltage is good then the OLED is somehow damaged. Although, I would say it's difficult to kill the OLED when soldering at high temperature when I see that your soldering skills are good.

Does the OLED work when you load the test firmware (Keyball44_test_v1.3.2)?

**Reply by maikeru0212** [reply]
OLEDとMCUの導通を確認しましたが、問題はありませんでした。
VCCとの電圧を確認しましたがDATAとRAWと電圧が3.3Vより低い結果となりました。

ちなみにテストファームウェアでもOLEDは動作しません。

I checked continuity between the OLED and MCU, and there was no problem.
I checked the voltage with VCC, but the voltage with DATA and RAW was lower than 3.3V.

By the way, OLED does not work even with the test firmware.

**Reply by jtuyen** [reply]
What is the VCC voltage though? It should be above 3.3V (mine is around 4.4V). You can also try swapping the left side OLED to the right side to see if it's functional.

**Reply by maikeru0212** [reply]
右側のOLEDを左側に実装したら、点灯しました。
新しくOLEDを注文して取り付けてみます。

数々の助言ありがとうございました！

When I installed the OLED on the right side on the left side, it lit up.
I'm going to order a new OLED and install it.

Thank you for all the advice!

**Comment by Yowkees** [comment]
Dear jtuyen, thank you very mush for your advice!

maikeru0212様ご連絡が遅くなり申し訳ありませんでした。
半田付けは非常にキレイにできていますので、OLEDが原因と分かって良かったです。
ぜひ完成させていただけることを願っています。
